book/getting-started/run-a-wallaroo-application-docker.md

```diff
@@ -0,0 +1,201 @@
+
+## Terminal 4, Run the "Celsius to Fahrenheit" Application
+
+Enter the Wallaroo Docker container:
+
+```bash
+docker exec -it wally environment-setup.sh
+```
+
+First, we'll need to get to the python Celsius example directory with the following command:
+
+```bash
+cd /src/wallaroo/examples/python/celsius
+```
+
+Now that we are in the proper directory, and the metrics UI and a data receiver are up and running, we can run the application itself by executing the following command:
+
+```bash
+machida --application-module celsius --in 127.0.0.1:7000 \
+  --out 127.0.0.1:5555 --metrics 127.0.0.1:5001 --control 127.0.0.1:6000 \
+  --data 127.0.0.1:6001 --name worker-name --external 127.0.0.1:5050 \
+  --cluster-initializer --ponythreads=1 --ponynoblock
+```
+
+This tells the "Celsius to Fahrenheit" application that it should listen on port `7000` for incoming data, write outgoing data to port `5555`, and send metrics data to port `5001`.
+
+## Terminal 5
+
+Enter the Wallaroo Docker container:
+
+```bash
+docker exec -it wally environment-setup.sh
+```
+
+### Sending Data with Giles Sender
+
+We will be sending in 25,000,000 messages using a pre-generated data file. The data file will be repeatedly sent via Giles Sender until we reach 25,000,000 messages.
+
+You will now be able to start the `sender` with the following command:
+
+```bash
+sender --host 127.0.0.1:7000 --messages 25000000 --binary --batch-size 300 \
+  --repeat --no-write --msg-size 8 --ponythreads=1 --ponynoblock \
+  --file /src/wallaroo/examples/python/celsius/celsius.msg
+```
+
+If the sender is working correctly, you should see `Connected` printed to the screen. If you see that, you can be assured that we are now sending data into our example application.
+
+## Check Out Some Metrics
+
+### First Look
+
+Once the sender has successfully connected, if you [visit the Metrics UI](http://localhost:4000) the landing page should show you that the "Celsius to Fahrenheit" application has successfully connected.
+
+![Landing Page](/book/metrics/images/landing-page.png)
+
+If your landing page resembles the one above, the "Celsius to Fahrenheit" application has successfully connected to the Metrics UI.
+
+Now, let's have a look at some metrics. By clicking on the "Celsius to Fahrenheit" link, you'll be taken to the "Application Dashboard" page. On this page you should see metric stats for the following:
+
+- a single pipeline: `Celsius Conversion`
+- a single worker: `Initializer`
+- three computations: `Add32`, `Decode Time in TCP Source`, `Multiply by 1.8`
+
+![Application Dashboard Page](/book/metrics/images/application-dashboard-page.png)
+
+You'll see the metric stats update as data continues to be processed in our application.
+
+You can then click into one of the elements within a category, to get to a detailed metrics page for that element. If we were to click into the `Add32` computation, we'll be taken to this page:
```

Comment:
"Click on an element within a category to get a..."
"If you click on the `Add32`..., you will be..."